tool/dumper/vkgcPipelineDumper.cpp

```diff
@@ -1067,13 +1066,21 @@ void PipelineDumper::updateHashForFragmentState(const GraphicsPipelineBuildInfo
 //
 // @param options: Pipeline options
 // @param [in/out] hasher : Hasher to generate hash code
-void PipelineDumper::updateHashForPipelineOptions(const PipelineOptions *options, MetroHash64 *hasher,
-                                                  bool isRelocatableShader) {
-  hasher->Update(options->includeDisassembly);
+// @param isCacheHash : True if the hash will be used as a key for a cache lookup.
+// @param isRelocatableShader : TRUE if we are building a relocatable shader
+void PipelineDumper::updateHashForPipelineOptions(const PipelineOptions *options, MetroHash64 *hasher, bool isCacheHash,
+                                                  bool isRelocatableShader, UnlinkedShaderStage stage) {
+  if (isCacheHash) {
```

Comment:
Do you think for safety reason, we could always update the hash unconditionally and remove all "if" checks?

Reply:
I could remove this one.  This will make sure stable hash is the same.  I don't have a strong feeling on these.

However, I cannot remove all "if" checks.  The the option does not effect something, then it could cause more cache misses.  I'm particularly concerned about relocatable shaders.  I explicitly want a cache hit if the only thing that is different is the shadow descriptor table usage.  I don't want to vertex processing stage to miss because the fragment shader enabled or disables interpModePatch.

Reply:
Thanks for explaining this. I think it's OK to keep it as-is.